tests/dashbio_demos/dash-ngl/app.py

```diff
@@ -0,0 +1,223 @@
+                                    )
+                                ],
+                                style={"width": "100%",
+                                       "display": "inline-block"},
+                            ),
+                        ],
+                    ),
+                    dcc.Tab(
+                        label="View",
+                        children=[
+                            html.Div(
+                                style={"height": "calc(98vh - 115px)"},
+                                children=[
+                                    # Dropdown to select the camera settings
+                                    # (perspective, orthographic
+                                    html.Div(["Camera settings"]),
+                                    dcc.Dropdown(
+                                        id="stage-camera-type",
+                                        options=[
+                                            {"label": k.capitalize(),
+                                             "value": k}
+                                            for k in ["perspective",
+                                                      "orthographic"]
+                                        ],
+                                        value="perspective",
+                                    ),
+                                    html.Div(["Background"]),
+                                    # Dropdown to select the background
+                                    # (white, black)
+                                    dcc.Dropdown(
+                                        id="stage-bg-color",
+                                        options=[
+                                            {"label": c, "value": c.lower()}
+                                            for c in ["black", "white"]
+                                        ],
+                                        value="white",
+                                    ),
+                                ],
+                            )
+                        ],
+                    ),
+                ],
+            ),
+        ]
+    )
+
+
+# Function to load the data from the folder data/
+def get_data(selection, pdb_id, color):
+    chain = "ALL"
+
+    # Check if only one chain should be shown
+    if "." in pdb_id:
+        pdb_id, chain = pdb_id.split(".")
+
+    fname = [f for f in glob.glob("data/" + pdb_id + ".*")][0]
+
+    ext = fname.split(".")[-1]
+    with open(fname, "r") as f:
+        contents = f.read()
+
+    return {
+        "selectedValue": selection,
+        "chain": chain,
+        "color": color,
+        "filename": fname.split("/")[-1],
+        "ext": ext,
+        "config": {"type": "text/plain", "input": contents},
+    }
+
+
+def callbacks(_app):
+
+    # Callback for molecule visualization based on the dropdown selection
+    @_app.callback(
+        [Output("viewport", "data")],
```

Comment:
```suggestion
        Output("viewport", "data"),
```
Very subtle error here -- if you wrap `Output` in a list like you've done here, Dash is expecting you to return a list or tuple of values from the callback that have a one-to-one correspondence with the items in the list. I see that `data` is also a list, but it's actually the value you want to *return* -- so in this case, you would have to return something like `[data]` or `(data,)`. It's easier (and makes more sense) to just drop the square brackets here.